addons/l10n_ke_edi_oscu/models/product.py

```diff
@@ -0,0 +1,361 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import logging
+import requests
+from collections import defaultdict
+
+from odoo import _, api, Command, fields, models
+from odoo.addons.base.models.ir_qweb_fields import Markup
+from odoo.tools.float_utils import json_float_round
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+    # TODO - reconcile field naming convention (same problem throughout, in account move etc)
+    #      - help strings for fields that appear on the product view
+    l10n_ke_packaging_unit_id = fields.Many2one(
+        'l10n_ke_edi_oscu.code',
+        related='product_variant_ids.l10n_ke_packaging_unit_id',
+        readonly=False, string='Packaging Unit',
+        domain=[('code_type', '=', '17')],
+        help='KRA code that describes the type of packaging used.',
+        copy=True,
+    )
+    l10n_ke_packaging_quantity = fields.Float(
+        string='Package Quantity',
+        related='product_variant_ids.l10n_ke_packaging_quantity',
+        readonly=False, default=1,
+        copy=True,
+    )
+    l10n_ke_origin_country_id = fields.Many2one(
+        'res.country',
+        related='product_variant_ids.l10n_ke_origin_country_id',
+        readonly=False, string='Origin Country',
+        copy=True,
+    )
+    l10n_ke_tax_type_code = fields.Char(related='product_variant_ids.l10n_ke_tax_type_code')
+    l10n_ke_unspsc_type_code = fields.Char(related='product_variant_ids.l10n_ke_unspsc_type_code')
+    l10n_ke_product_type_code = fields.Selection(
+        related='product_variant_ids.l10n_ke_product_type_code',
+        readonly=False,
+        copy=True,
+    )
+    l10n_ke_is_insurance_applicable = fields.Boolean(
+        related='product_variant_ids.l10n_ke_is_insurance_applicable',
+        readonly=False,
+        copy=True,
+    )
+    l10n_ke_item_code = fields.Char(string="KRA Item Code", related='product_variant_ids.l10n_ke_item_code')
+
+    def action_l10n_ke_oscu_save_item(self):
+        if self.product_variant_count != 1:
```

Comment:
The button will be visible on the product.template view if there is only one variant. 
It's useful, because if variants aren't enabled, the user can access the product.template view but not the product.product view.